src/broadcast/participant.rs

```diff
@@ -83,8 +83,22 @@ pub(crate) struct BroadcastIndex {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct BroadcastOutput {
-    pub(crate) tag: BroadcastTag,
-    pub(crate) msg: Message,
+    tag: BroadcastTag,
+    msg: Message,
+}
+
+impl BroadcastOutput {
+    pub(crate) fn extract_message(&self, expected_tag: BroadcastTag) -> Result<&Message> {
```

Comment:
suggestion: can this take self by value instead of by reference? I think in most cases extracting the message is the last step of processing a broadcast output so we can consume the memory.

Reply:
If this works, you can rename the function to `into_message`, following [Rust's naming conventions for conversion functions](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv).